src/mtcp/mtcp_restart.c

```diff
@@ -235,12 +238,18 @@ main(int argc, char *argv[], char **environ)
 #endif /* ifdef ENABLE_VDSO_CHECK */
 
   rinfo.fd = -1;
+  rinfo.mtcp_restart_pause = 0; /* false */
   rinfo.use_gdb = 0;
   rinfo.text_offset = -1;
   shift;
   while (argc > 0) {
+    // All parsing of flags with no arg must appear before parsing ckptImage
+    if (mtcp_strcmp(argv[0], "--mtcp-restart-pause") == 0) {
```

Comment:
Nice point!  The `argc == 1` test should come after any flags with no arguments, and that's donce most easily if it comes last.  I'll move it to the end, and place thje flags with no arguments near the end.